app/controllers/concerns/admin_only_maintenance_mode.rb

```diff
@@ -0,0 +1,19 @@
+module AdminOnlyMaintenanceMode
+  extend ActiveSupport::Concern
+  include ActionView::Helpers::TextHelper
+
+  included do
+    before_action :redirect_non_admins_in_maintenance_mode
+  end
+
+private
+
+  def redirect_non_admins_in_maintenance_mode
+    return unless Settings.features.maintenance_mode.enabled
+    return if %w[sign_in sessions].include?(controller_name)
+
+    unless current_user && current_user["admin"]
```

Comment:
Can we extract this to a method... `redirect_if_non_admin` or something. Codeclimate is upset.